### PR TITLE
Review pg_autoctl show state output, and docs.

### PR DESCRIPTION
**Pull request by DimCitus**
Make it clear which kind of connection is given by which node, in particular
that when in the SINGLE or WAIT_PRIMARY state we are still delivering a
read-write connection to the application.

The "Reachable" column is now a "Connection" column and the node health as
seen by the monitor is encoded in the single-char *, !, or ?, depending on
the health value (1, 0, -1).

See #608.

With this PR applied we can get such output:

```
$ pg_autoctl show state
 Name |  Node |      Host:Port |       LSN |   Connection |       Current State |      Assigned State
------+-------+----------------+-----------+--------------+---------------------+--------------------
node1 |     1 | localhost:5501 | 0/40001F8 | read-write * |             primary |             primary
node2 |     2 | localhost:5502 | 0/40001F8 |  read-only * |           secondary |           secondary
node3 |     3 | localhost:5503 | 0/40001F8 |       none * |         maintenance |         maintenance
```

Another example involving a `wait_primary` node:

```
$ pg_autoctl show state
 Name |  Node |      Host:Port |       LSN |   Connection |       Current State |      Assigned State
------+-------+----------------+-----------+--------------+---------------------+--------------------
node1 |     1 | localhost:5501 | 0/3000000 | read-write * |        wait_primary |        wait_primary
node2 |     2 | localhost:5502 |       0/0 |       none ? |        wait_standby |          catchingup
```

### CODE REVIEW

File: src/bin/pg_autoctl/nodestate_utils.c [hunk 1]
```diff
@@ -382,6 +391,92 @@ nodestateHealthToString(int health)
...
+		}
+
+		case 0:
+		{
+			return '!';
+		}
+
+		case 1:
+		{
+			return '*';
```
Comment (JelteF) [inline src/bin/pg_autoctl/nodestate_utils.c:414]:
```suggestion
			return '✓';
```

File: src/bin/pg_autoctl/nodestate_utils.c [hunk 2]
```diff
@@ -382,6 +391,92 @@ nodestateHealthToString(int health)
...
+	switch (health)
+	{
+		case -1:
+		{
+			return '?';
+		}
+
+		case 0:
+		{
+			return '!';
```
Comment (JelteF) [inline src/bin/pg_autoctl/nodestate_utils.c:409]:
```suggestion
			return '✗';
```

**Review by JelteF — APPROVED** [review]
I think these symbols are a bit clearer (especially the checkmark vs the *)

File: src/bin/pg_autoctl/nodestate_utils.c (hunk 1 above)
Comment (JelteF) [reply src/bin/pg_autoctl/nodestate_utils.c:414]: Maybe this being a space as also totally fine, i.e. no need to have a character shown if everything is good.

Comment (DimCitus) [reply src/bin/pg_autoctl/nodestate_utils.c]: I don't know how to maintain table alignment when using wider unicode chars, and unfortunately both ✓ and ✗ are wider than space.

File: src/bin/pg_autoctl/nodestate_utils.c (hunk 2 above)
Comment (DimCitus) [reply src/bin/pg_autoctl/nodestate_utils.c:415]: Same unicode remark with wider-than-space characters and table alignment.

**Pull request merged**